api/routes/websoc/index.ts

```diff
@@ -1,572 +1,151 @@
-                );
-                logger.warn(`Stack trace: ${e}`);
-                continue;
-              }
-            }
-            if (q.sectionCodes) {
-              const sectionCodes = q.sectionCodes.split(",");
-              try {
-                const items = await Promise.all(
-                  sectionCodes.map((sectionCode) =>
-                    docClient
-                      .query(
-                        tableName,
-                        {
-                          name: "requestHash",
-                          value: hash([
-                            term,
-                            { sectionCodes: sectionCode } as WebsocAPIOptions,
-                          ]),
-                        },
-                        sortKey
-                      )
-                      .then((x) => x?.Items)
-                  )
-                );
-                if (items.every((x) => x?.length)) {
-                  logger.info(`Cache hit: ${JSON.stringify(q)}`);
-                  queries[parseInt(i)] = undefined;
-                  ret = combineResponses(
-                    ...items.map(
-                      (x): WebsocAPIResponse => x?.slice(-1)[0].data
-                    ),
-                    ret
-                  );
-                } else {
-                  logger.info(`Cache miss: ${JSON.stringify(q)}`);
-                  sectionCodes.map((sectionCode) =>
-                    dispatchCacheUpdater(lambdaClient, tableName, term, {
-                      sectionCodes: sectionCode,
-                    })
-                  );
-                }
-              } catch (e) {
-                logger.warn(
-                  `Error occurred while querying cache for ${JSON.stringify(q)}`
-                );
-                logger.warn(`Stack trace: ${e}`);
-              }
-            }
+          if (websocSections.length) {
+            const websocApiResponses = websocSections
+              .map((x) => x.data)
+              .filter(notNull) as WebsocAPIResponse[];
+            const combinedResponses = combineResponses(...websocApiResponses);
+            return createOKResult(sortResponse(combinedResponses), requestId);
           }
-          // Filter out the queries that have been set to ``undefined``,
-          // i.e. the ones we have fulfilled through the cache already.
-          queries = queries.filter((q) => q);
         }
-        // For each remaining query:
-        // Collect them into an iterable of ``Promises`` and fire them in parallel.
-        // Check if each result fulfilled or rejected; merge the data of each
-        // fulfilled response into the final response and remove it from the
-        // array of queries. If there are still responses remaining, wait for 1
-        // second and repeat the process.
+
+        let queries = normalizeQuery(parsedQuery);
+        let websocResponseData: WebsocAPIResponse = { schools: [] };
+        let retries = 0;
+
```

Comment:
```suggestion
let failed: WebsocAPIOptions[] = [];
```
We can move the failed array outside as well.